elytron/ELY-2078-encryption-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,214 @@
+The identity file being encrypted with a SecretKey that contains a bcrypt hashed password along with 5 attributes would look like the following.
+
+```XML
+<?xml version="1.0" ?>
+<identity xmlns="urn:elytron:1.0">
+    <credentials>
+        <password algorithm="bcrypt" format="enc_base64">RUxZAUMQo2svngST8+WfAg6BoeBtBH0exXBlwAQvN4jCxEuDwPUkniZkgx/3mfetwDJK4rql0jfaigStMAW8mE9iNXG5QQ==</password>
+    </credentials>
+    <attributes>
+        <attribute name="RUxZAUMQUO9rq4PRc3BB61B5oZSepATonjzj1dlUriqbsy//5fw=" value="RUxZAUMQwbcAyojI0gsU0BFxeuBw/zJXR8Umy/jk9tnWrvpRxu0="></attribute>
+        <attribute name="RUxZAUMQ5iZ43cvM0x/RwPxwnqxXZisDmFX0Un+f3pLQz0WDC3o=" value="RUxZAUMQWUAYVdzpdv4sKGaFblcIMsso6x0j5J9oCSHQiDbjJE8="></attribute>
+        <attribute name="RUxZAUMQl1xAhDoJdsMnk7ggw66zyClhUzzB79p859BkQrXazzo=" value="RUxZAUMQCV32L5vunNfLMCISdjyn7myJ0saJdAWreKq1BjnGjfU="></attribute>
+        <attribute name="RUxZAUMQ7t3uOr63lNMwD+Ljsq68ONPwGYdtbVLtdmbY6stZIjg=" value="RUxZAUMQ41IKUmpazIiXd4wQa67I+y35TLtssweZ/1aci7w71ag="></attribute>
+        <attribute name="RUxZAUMQDIjiDvRMr9sOCHnCyh8/uwytBJZOfgCsH0k0itNwcNA=" value="RUxZAUMQi3C21O9+je0b9kXHIsQyoiuSYYafYAFKS3h7iPM9+OE="></attribute>
+    </attributes></identity>
+```
+=== Integrity of Filesystem Realm
+The two approaches being considered are between signing each identity file, and just calculating the checksum of each identity file.
+
+The advantage of using just a straightforward checksum using a hash algorithm is that it will not be resource heavy, as an algorithm such as SHA256 or MD5 is very efficient, although it comes with a caveat that it is possible to produce collisions (false positivies) due to the lack of complexity of the algorithm. The disadvantage of using a simple checksum is that if an authorized user were to gain access to the master list of checksums that is used to verify each identity when accessed, they could change the checksum rendering that identity to be useless, until that checksum is amended. Is it also possible for the unauthorized user to modify the identity file, calculate the new checksum, and update the checksum list with an updated one. This would be very detrimental as we have no way of knowing that the file has been modified and is no longer valid.
+
+If the approach of signing a file was chosen, it would require an asymmetric key pair, a public key and secret key. This would involve another credential store and a method to ensure only the correct party has access to the secret key.
+
+It could be made that the public key is available for everyone to use. This means anyone can decrypt the signature to get the hash of the identity, but only people with access to the secret key for the encryption of the realm would have access to the private key that encrypts the hash of each identity. This way an unauthorized user could view the hash via the public key, but cannot change the hash/signature of the identity.
```

Comment:
"only people with access to the secret key for the encryption of the realm would have access to the private key that encrypts the hash of each identity"

->

"only people with access to the key store for the realm would have access to the private key..."

Reply:
Updated